src/text/text.rs

```diff
@@ -7,49 +7,102 @@ use crate::{
         Alignment, Baseline, TextStyle,
     },
     transform::Transform,
-    Drawable, SaturatingCast, Styled,
+    Drawable, SaturatingCast,
 };
+
+use super::TextStyleBuilder;
 /// A text object.
 ///
 /// The `Text` struct represents a string that can be drawn onto a display.
 ///
+/// TODO: The paragraph below needs to be updated, because `Text` no longer requires `Styled`.
+///
 /// The text object only contains the string and position and no additional styling information,
 /// like the font or color. To draw a text object it is necessary to attach a style to it by using
-/// the [`into_styled`] method to create a [`Styled`] object.
+/// the `into_styled` method to create a `Styled` object.
 ///
 /// See the [module-level documentation] for examples how to use text objects.
 ///
-/// [`into_styled`]: #method.into_styled
-/// [`Styled`]: ../struct.Styled.html
 /// [module-level documentation]: index.html
 #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash, Debug)]
-pub struct Text<'a> {
+pub struct Text<'a, S> {
     /// The string.
     pub text: &'a str,
 
     /// The position.
     ///
     /// Defines the top-left starting pixel of the text object.
     pub position: Point,
+
+    /// The character style.
+    pub character_style: S,
+
+    /// The text style.
+    pub text_style: TextStyle,
 }
 
-impl<'a> Text<'a> {
+impl<'a, S> Text<'a, S> {
     /// Creates a text.
-    pub const fn new(text: &'a str, position: Point) -> Self {
-        Self { text, position }
+    pub const fn new(text: &'a str, position: Point, character_style: S) -> Self {
+        Self {
+            text,
+            position,
+            character_style,
+            text_style: TextStyleBuilder::new().build(),
+        }
     }
 
-    /// Attaches a text style to the text object.
-    pub fn into_styled<S>(self, style: S) -> Styled<Self, S> {
-        Styled::new(self, style)
+    /// Creates a text.
+    pub const fn with_text_style(
+        text: &'a str,
+        position: Point,
+        character_style: S,
+        text_style: TextStyle,
+    ) -> Self {
+        Self {
+            text,
+            position,
+            character_style,
+            text_style,
+        }
+    }
+
+    /// Creates a text.
+    pub const fn with_baseline(
+        text: &'a str,
+        position: Point,
+        character_style: S,
+        baseline: Baseline,
+    ) -> Self {
+        Self {
+            text,
+            position,
+            character_style,
+            text_style: TextStyle::with_baseline(baseline),
+        }
+    }
+
+    /// Creates a text.
```

Comment:
I think it's horizontal?

```suggestion
    /// Creates a text with given horizontal alignment.
```